src/App.tsx

```diff
@@ -1,27 +1,90 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { fetchData, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [data, setData]
+   = useState<Array<{ id: number, name: string, color: string }>>([]);
+  const [name, setName] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  async function getData(value: string) {
+    let fetchedData;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    switch (value) {
+      case 'everyItem':
+        fetchedData = await fetchData();
+        break;
+      case 'first5':
+        fetchedData = await get5First();
+        break;
+      case 'onlyRed':
+        fetchedData = await getRedGoods();
+        break;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      default:
+        fetchedData = await fetchData();
+    }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setData(fetchedData);
+  }
+
+  useEffect((value = 'everyItem') => {
+    getData(value);
+  }, []);
+
+  const reRender = (value: string) => {
+    getData(value);
+  };
```

Comment:
Do we need this extra function `reRender`?

Reply:
@Peti51 
Don't leave unresolved comments, please. 
If you have any questions, you can ask the comment author or just ask in Slack.